Core.Collectors/Collector/CollectorBase.cs

```diff
@@ -63,8 +63,18 @@ public async Task<bool> ProcessAsync(T collectionNode, long maxPageCount)
             while (batchingHttpRequest.HasNext && counter < maxPageCount && !haltCollection)
             {
                 counter++;
-                HttpResponseMessage response = await batchingHttpRequest.NextResponseAsync(this.authentication).ConfigureAwait(false);
+                RequestResult result = await batchingHttpRequest.NextResponseAsync(this.authentication).ConfigureAwait(false);
 
+                if (!result.IsSuccess())
+                {
+                    foreach(T allowListCollectionNode in result.allowListStatus.Continuation())
```

Comment:
>T [](http://example.com/codeflow?start=28&length=1)

Are we certain that this cast is fine? Continuation() here returns a CollectionNode, so in theory, this could fail, no?

Reply:
Address this in new commit and opened a new comment.